packages/lesswrong/components/hooks/usePaginatedResolver.ts

```diff
@@ -94,10 +95,10 @@ export const usePaginatedResolver = <
   }
 
   useEffect(() => {
-    if (results?.length && results.length >= (list?.length ?? 0)) {
+    if (results?.length && results.length >= (list?.length ?? 0) && !isEqual(list, results)) {
```

Comment:
This change was motivated by an early attempt to use `usePostsList`, but even without that, I think it's still an improvement (since it now no longer causes unnecessary renders on e.g. a "load more" which doesn't actually end up pulling more records).